src/main/kotlin/com/hibob/types/StoreService.kt

```diff
@@ -0,0 +1,65 @@
+package com.hibob.types
+
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+
+@Component
+class StoreService {
+
+    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
+
+        return cart.associate { cart ->
+            cart.clientId to checkOut(cart, payment) }
+    }
+
+    fun checkOut(cart : Cart, payment: Payment): Check {
+
+        val totalPrice = calculateTotalPrice(listOf(cart))
+
+        val status = isPaymentValid(payment, totalPrice, LocalDate.now())
+
+        val finalPrice = if (status == Statuses.SUCCESS) totalPrice else 0.0
+
+        return Check(cart.clientId, status, finalPrice)
+    }
+
+    private fun fail(message: String): Nothing {
+        throw IllegalStateException(message)
+    }
+
+    private fun checkProduct(product: Product): Boolean {
+        return product.custom == true
+    }
+
+    private fun calculateTotalPrice(cart: List<Cart>): Double {
```

Comment:
why to get list if you always send one cart? using cart: Cart make more sense.
think about it you create a list in order to send it to the function and then use flatMap, I understand you want to do it generic function, but no need in this case especially if its a private method.